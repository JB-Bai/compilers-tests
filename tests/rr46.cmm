int bqnXrxKW(){
	float ds3QrZGee[57] = MmGLduyF2I(vw4tiB) = VXqAIE || NqdqiG.CzzaC0g_C[s29u_K131937(- mtIc2u() || dgZvLL65HM6((9.3704), 004 + 91.9768E7 = EBDa0JtGg() + - wVqPkJ9 && WcAzA2d.gIkMz || xCx8()) || QDCZZTuhs() / ZuS / 07.316E+9 >= 0722 + 4.3 / 073121 - 823.2 - 7786 - cw06XWp(9794.415) && 02072 = 126.9 - 06131[BmZdbpde5K()] >= PdOyLTX = IBirF5() / jh8K8f, 0xb5)], PQQWqJamr =(5764.9599).wOkWs1, LxkJbryZM;
	if(N74Z8k5Z(949)) ZA_4JWDUZ5P();
}
float hTM15rD;
int im9RLpPp1X(struct nTs3FC H4kTN, int _BEvj[0xc][0x3f2][02613], struct{
	struct wvC4dI{
	}
	C_yzWOiQ2 = k6UpemnZN < ! _Q0nLYL.w3Sow5jpI8, dZIVDCFC9U = ! UcT9Ef, bnaqOlJD5 = 31;
	int z5FiKWa[06634];
	float dRHrNxar7 = 764.994E-2[0xE2] -(ok7uvSI0hU) || 85.36 || 0xA44[((38.139e9))][0267] - jsghyhyb.A_89eEaStg / ! Sf2Kodfq3o() - 12.915;
	int ks_9ehHmd = - 0561 || - W7W_i9pVo;
	int d7O6jJn, xmkUaCORJ = 0xE9c * uiJyO7u(), fLToFu70 = 358.624, L_1UMED[0xc], BfEpM[388];
}
I9jn75V){
	dcsYuuT(DRr8PcaHa).bmwUsERq;
}

