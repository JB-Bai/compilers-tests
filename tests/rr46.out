Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: bqnXrxKW
        LP
        RP
      CompSt (1)
        LC
        DefList (2)
          Def (2)
            Specifier (2)
              TYPE: float
            DecList (2)
              Dec (2)
                VarDec (2)
                  VarDec (2)
                    ID: ds3QrZGee
                  LB
                  INT: 57
                  RB
                ASSIGNOP
                Exp (2)
                  Exp (2)
                    ID: MmGLduyF2I
                    LP
                    Args (2)
                      Exp (2)
                        ID: vw4tiB
                    RP
                  ASSIGNOP
                  Exp (2)
                    Exp (2)
                      ID: VXqAIE
                    OR
                    Exp (2)
                      Exp (2)
                        Exp (2)
                          ID: NqdqiG
                        DOT
                        ID: CzzaC0g_C
                      LB
                      Exp (2)
                        ID: s29u_K131937
                        LP
                        Args (2)
                          Exp (2)
                            Exp (2)
                              Exp (2)
                                Exp (2)
                                  MINUS
                                  Exp (2)
                                    ID: mtIc2u
                                    LP
                                    RP
                                OR
                                Exp (2)
                                  ID: dgZvLL65HM6
                                  LP
                                  Args (2)
                                    Exp (2)
                                      LP
                                      Exp (2)
                                        FLOAT: 9.370400
                                      RP
                                    COMMA
                                    Args (2)
                                      Exp (2)
                                        Exp (2)
                                          Exp (2)
                                            INT: 4
                                          PLUS
                                          Exp (2)
                                            FLOAT: 919768000.000000
                                        ASSIGNOP
                                        Exp (2)
                                          Exp (2)
                                            Exp (2)
                                              Exp (2)
                                                ID: EBDa0JtGg
                                                LP
                                                RP
                                              PLUS
                                              Exp (2)
                                                MINUS
                                                Exp (2)
                                                  ID: wVqPkJ9
                                            AND
                                            Exp (2)
                                              Exp (2)
                                                ID: WcAzA2d
                                              DOT
                                              ID: gIkMz
                                          OR
                                          Exp (2)
                                            ID: xCx8
                                            LP
                                            RP
                                  RP
                              OR
                              Exp (2)
                                Exp (2)
                                  Exp (2)
                                    Exp (2)
                                      Exp (2)
                                        ID: QDCZZTuhs
                                        LP
                                        RP
                                      DIV
                                      Exp (2)
                                        ID: ZuS
                                    DIV
                                    Exp (2)
                                      FLOAT: 7315999744.000000
                                  RELOP
                                  Exp (2)
                                    Exp (2)
                                      Exp (2)
                                        Exp (2)
                                          Exp (2)
                                            INT: 466
                                          PLUS
                                          Exp (2)
                                            Exp (2)
                                              FLOAT: 4.300000
                                            DIV
                                            Exp (2)
                                              INT: 30289
                                        MINUS
                                        Exp (2)
                                          FLOAT: 823.200012
                                      MINUS
                                      Exp (2)
                                        INT: 7786
                                    MINUS
                                    Exp (2)
                                      ID: cw06XWp
                                      LP
                                      Args (2)
                                        Exp (2)
                                          FLOAT: 9794.415039
                                      RP
                                AND
                                Exp (2)
                                  INT: 1082
                            ASSIGNOP
                            Exp (2)
                              Exp (2)
                                Exp (2)
                                  Exp (2)
                                    FLOAT: 126.900002
                                  MINUS
                                  Exp (2)
                                    Exp (2)
                                      INT: 3161
                                    LB
                                    Exp (2)
                                      ID: BmZdbpde5K
                                      LP
                                      RP
                                    RB
                                RELOP
                                Exp (2)
                                  ID: PdOyLTX
                              ASSIGNOP
                              Exp (2)
                                Exp (2)
                                  ID: IBirF5
                                  LP
                                  RP
                                DIV
                                Exp (2)
                                  ID: jh8K8f
                          COMMA
                          Args (2)
                            Exp (2)
                              INT: 181
                        RP
                      RB
              COMMA
              DecList (2)
                Dec (2)
                  VarDec (2)
                    ID: PQQWqJamr
                  ASSIGNOP
                  Exp (2)
                    Exp (2)
                      LP
                      Exp (2)
                        FLOAT: 5764.959961
                      RP
                    DOT
                    ID: wOkWs1
                COMMA
                DecList (2)
                  Dec (2)
                    VarDec (2)
                      ID: LxkJbryZM
            SEMI
        StmtList (3)
          Stmt (3)
            IF
            LP
            Exp (3)
              ID: N74Z8k5Z
              LP
              Args (3)
                Exp (3)
                  INT: 949
              RP
            RP
            Stmt (3)
              Exp (3)
                ID: ZA_4JWDUZ5P
                LP
                RP
              SEMI
        RC
    ExtDefList (5)
      ExtDef (5)
        Specifier (5)
          TYPE: float
        ExtDecList (5)
          VarDec (5)
            ID: hTM15rD
        SEMI
      ExtDefList (6)
        ExtDef (6)
          Specifier (6)
            TYPE: int
          FunDec (6)
            ID: im9RLpPp1X
            LP
            VarList (6)
              ParamDec (6)
                Specifier (6)
                  StructSpecifier (6)
                    STRUCT
                    Tag (6)
                      ID: nTs3FC
                VarDec (6)
                  ID: H4kTN
              COMMA
              VarList (6)
                ParamDec (6)
                  Specifier (6)
                    TYPE: int
                  VarDec (6)
                    VarDec (6)
                      VarDec (6)
                        VarDec (6)
                          ID: _BEvj
                        LB
                        INT: 12
                        RB
                      LB
                      INT: 1010
                      RB
                    LB
                    INT: 1419
                    RB
                COMMA
                VarList (6)
                  ParamDec (6)
                    Specifier (6)
                      StructSpecifier (6)
                        STRUCT
                        LC
                        DefList (7)
                          Def (7)
                            Specifier (7)
                              StructSpecifier (7)
                                STRUCT
                                OptTag (7)
                                  ID: wvC4dI
                                LC
                                RC
                            DecList (9)
                              Dec (9)
                                VarDec (9)
                                  ID: C_yzWOiQ2
                                ASSIGNOP
                                Exp (9)
                                  Exp (9)
                                    ID: k6UpemnZN
                                  RELOP
                                  Exp (9)
                                    NOT
                                    Exp (9)
                                      Exp (9)
                                        ID: _Q0nLYL
                                      DOT
                                      ID: w3Sow5jpI8
                              COMMA
                              DecList (9)
                                Dec (9)
                                  VarDec (9)
                                    ID: dZIVDCFC9U
                                  ASSIGNOP
                                  Exp (9)
                                    NOT
                                    Exp (9)
                                      ID: UcT9Ef
                                COMMA
                                DecList (9)
                                  Dec (9)
                                    VarDec (9)
                                      ID: bnaqOlJD5
                                    ASSIGNOP
                                    Exp (9)
                                      INT: 31
                            SEMI
                          DefList (10)
                            Def (10)
                              Specifier (10)
                                TYPE: int
                              DecList (10)
                                Dec (10)
                                  VarDec (10)
                                    VarDec (10)
                                      ID: z5FiKWa
                                    LB
                                    INT: 3484
                                    RB
                              SEMI
                            DefList (11)
                              Def (11)
                                Specifier (11)
                                  TYPE: float
                                DecList (11)
                                  Dec (11)
                                    VarDec (11)
                                      ID: dRHrNxar7
                                    ASSIGNOP
                                    Exp (11)
                                      Exp (11)
                                        Exp (11)
                                          Exp (11)
                                            Exp (11)
                                              FLOAT: 7.649940
                                            LB
                                            Exp (11)
                                              INT: 226
                                            RB
                                          MINUS
                                          Exp (11)
                                            LP
                                            Exp (11)
                                              ID: ok7uvSI0hU
                                            RP
                                        OR
                                        Exp (11)
                                          FLOAT: 85.360001
                                      OR
                                      Exp (11)
                                        Exp (11)
                                          Exp (11)
                                            Exp (11)
                                              Exp (11)
                                                INT: 2628
                                              LB
                                              Exp (11)
                                                LP
                                                Exp (11)
                                                  LP
                                                  Exp (11)
                                                    FLOAT: 38138998784.000000
                                                  RP
                                                RP
                                              RB
                                            LB
                                            Exp (11)
                                              INT: 183
                                            RB
                                          MINUS
                                          Exp (11)
                                            Exp (11)
                                              Exp (11)
                                                ID: jsghyhyb
                                              DOT
                                              ID: A_89eEaStg
                                            DIV
                                            Exp (11)
                                              NOT
                                              Exp (11)
                                                ID: Sf2Kodfq3o
                                                LP
                                                RP
                                        MINUS
                                        Exp (11)
                                          FLOAT: 12.915000
                                SEMI
                              DefList (12)
                                Def (12)
                                  Specifier (12)
                                    TYPE: int
                                  DecList (12)
                                    Dec (12)
                                      VarDec (12)
                                        ID: ks_9ehHmd
                                      ASSIGNOP
                                      Exp (12)
                                        Exp (12)
                                          MINUS
                                          Exp (12)
                                            INT: 369
                                        OR
                                        Exp (12)
                                          MINUS
                                          Exp (12)
                                            ID: W7W_i9pVo
                                  SEMI
                                DefList (13)
                                  Def (13)
                                    Specifier (13)
                                      TYPE: int
                                    DecList (13)
                                      Dec (13)
                                        VarDec (13)
                                          ID: d7O6jJn
                                      COMMA
                                      DecList (13)
                                        Dec (13)
                                          VarDec (13)
                                            ID: xmkUaCORJ
                                          ASSIGNOP
                                          Exp (13)
                                            Exp (13)
                                              INT: 3740
                                            STAR
                                            Exp (13)
                                              ID: uiJyO7u
                                              LP
                                              RP
                                        COMMA
                                        DecList (13)
                                          Dec (13)
                                            VarDec (13)
                                              ID: fLToFu70
                                            ASSIGNOP
                                            Exp (13)
                                              FLOAT: 358.623993
                                          COMMA
                                          DecList (13)
                                            Dec (13)
                                              VarDec (13)
                                                VarDec (13)
                                                  ID: L_1UMED
                                                LB
                                                INT: 12
                                                RB
                                            COMMA
                                            DecList (13)
                                              Dec (13)
                                                VarDec (13)
                                                  VarDec (13)
                                                    ID: BfEpM
                                                  LB
                                                  INT: 388
                                                  RB
                                    SEMI
                        RC
                    VarDec (15)
                      ID: I9jn75V
            RP
          CompSt (15)
            LC
            StmtList (16)
              Stmt (16)
                Exp (16)
                  Exp (16)
                    ID: dcsYuuT
                    LP
                    Args (16)
                      Exp (16)
                        ID: DRr8PcaHa
                    RP
                  DOT
                  ID: bmwUsERq
                SEMI
            RC
